zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -402,6 +392,34 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
             newAlert.setWascId(wascId);
             extension.alertFound(newAlert, msg.getHistoryRef());
             return new ApiResponseElement(name, Integer.toString(newAlert.getAlertId()));
+        }  else if (ACTION_UPDATE_ALERTS_CONFIDENCE.equals(name)) {
+            String alertIds = ApiUtils.getNonEmptyStringParam(params, PARAM_ALERT_IDS);
+            int confidenceId = getConfidenceId(params);
+
+            StringTokenizer tokenizer = new StringTokenizer(alertIds, ",");
+            while (tokenizer.hasMoreTokens()) {
```

Comment:
- I'd suggest collect them first to ensure all IDs are valid before starting processing the alerts (while missing alert will also lead to partial updates it's one less thing).
 - This can also be extracted, something like:
`updateAlerts(alert -> alert.setConfidence(confidenceId));`
and let `updateAlerts` extract/loop the IDs and get/process the alerts.

Reply:
For the second one, how would the signature/definition of the method look?

Reply:
Right, left out the `params`... should be like:
`void processAlerts(JSONObject params, Consumer<Alert> consumer)`

Reply:
TY!